ovos_PHAL_plugin_system/__init__.py

```diff
@@ -62,11 +63,21 @@ def use_external_factory_reset(self):
         return external_requested or False
 
     def handle_reset_register(self, message):
+        if not message.data.get("skill_id"):
+            LOG.warning(f"Got registration request without a `skill_id`: "
```

Comment:
why would this happen? its a new bus message, nothing should be emitting it and we dont need backwards compat?

Reply:
the whole point of registering is to signal individual plugins need extra reset actions, such as a mk1 reflashing arduino for faceplate. its a totally new mechanism. so there should never ever be a case without skill id

Reply:
This is basically backwards-compat. for the shell and some handling to prevent throwing an exception in the unsafe reference after this